architecture.tex

```diff
@@ -87,10 +87,14 @@
 
```

Comment:
How about a section to describe the args invariant, the tree structure, and the fundamental classes.?